internal/semaphore/semaphore_test.go

```diff
@@ -0,0 +1,93 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package semaphore
+
+import (
+	"sync"
+	"testing"
+	"github.com/stretchr/testify/assert"
+)
+
+func checkLimit(t *testing.T, sem Semaphore, expected int) {
+	limit := sem.GetLimit()
+	assert.Equal(t, expected, limit)
+}
+
+func checkCount(t *testing.T, sem Semaphore, expected int) {
+	count := sem.GetCount()
+	assert.Equal(t, expected, count)
+}
+
+func checkLimitAndCount(t *testing.T, sem Semaphore, expectedLimit, expectedCount int) {
+	checkLimit(t, sem, expectedLimit)
+	checkCount(t, sem, expectedCount)
+}
+
+func TestNewSemaphore_NegativeLimit(t *testing.T) {
+	assert.Panics(t, func() { _ = NewSemaphore(-1) })
+}
+
+func TestSemaphore_AcquireRelease(t *testing.T) {
+	sem := NewSemaphore(2)
+	checkLimitAndCount(t, sem, 2, 0)
+
+	sem.Acquire()
+	checkLimitAndCount(t, sem, 2, 1)
+
+	sem.Acquire()
+	checkLimitAndCount(t, sem, 2, 2)
+
+	sem.Release()
+	checkLimitAndCount(t, sem, 2, 1)
+
+	sem.Release()
+	checkLimitAndCount(t, sem, 2, 0)
+}
+
+func TestSemaphore_ReleaseMoreThanAcquire(t *testing.T) {
+	sem := NewSemaphore(1)
```

Comment:
One of the ways that you are using the semaphore is creating one with a zero limit, `NewSemaphore(0)`. Do we need test cases for that?

Reply:
Yes. I have added [NewSemaphore(0)](https://github.com/aws/amazon-cloudwatch-agent/pull/491/files#diff-dabd66c7b47b8a390b2b5716a07ee02e854cf06cdb4a5288a1e10854b7d1ee19R43-R50). Thanks for reminding me